extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/AgroalBuildTimeConfig.java

```diff
@@ -20,4 +20,10 @@
      */
     @ConfigItem(name = ConfigItem.PARENT)
     public Map<String, DataSourceBuildTimeConfig> namedDataSources;
+
+    /**
+     * Whether or not an healtcheck is published in case the smallrye-health extionsion is present (default to true).
+     */
+    @ConfigItem(name = "health.enabled", defaultValue = "true")
```

Comment:
I wonder if leaving out this naming the config item and having a default name `health-enabled` would have been a viable alternative? 

Also would one of `enable-healthcheck` or `healthcheck-enabled` would be better name for this config item?

Should the default value be controlled by whatever is set in `HealthBuildTimeConfig#extensionsEnabled`?

Reply:
This is a design choice still opens to discussion ;) I know that @emmanuelbernard want that as much as possible works will be included inside the smallrye-health extension but I think that it's the agroal extension that needs to control it's configuration and the integration of the health check: this will be less works to do during the augmenation phase ...

Your proposal to add `HealthBuildTimeConfig#extensionsEnabled` will not facilitate the usage for the agroal extension a lot as it will still need to support the configuration item ...